filters/tags_filter.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package filters
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+)
+
+// SortedTagIterator iterates over a set of tag names and values
+// sorted by tag names in ascending order
+type SortedTagIterator interface {
+	// Next returns true if there are more tag names and values
+	Next() bool
+
+	// Current returns the current tag name and value
+	Current() (string, string)
+
+	// Err returns any errors encountered
+	Err() error
+
+	// Close closes the iterator
+	Close()
+}
+
+// NewSortedTagIteratorFn creates a tag iterator given an id
+type NewSortedTagIteratorFn func(id string) SortedTagIterator
+
+// tagFilter is a filter associated with a given tag
+type tagFilter struct {
+	name        string
+	valueFilter Filter
```

Comment:
I named it valueFilter to indicate that you will be filtering on the value only, not the name.